src/xdp/dispatch.c

```diff
@@ -291,7 +291,6 @@ XdpFastDeviceIoControl(
     PAGED_CODE();
 
     if (FileObjHeader == NULL) {
-        IoStatus->Status = STATUS_INVALID_PARAMETER;
```

Comment:
I can't see because of the `_Use_decl_annotations_`, is there a `_Success_` annotation only when `return == TRUE`?

Reply:
No, that's missing. That's why the output parameter warning is already suppressed for this function.

Reply:
IMO, adding the annotation would be better than suppressing.

Reply:
I just gave that a try, but it ended up making things messy: we then have to fully declare the routine at the top of the file, rather than just saying we have a `FAST_IO_DEVICE_CONTROL` routine. I'd agree with you in general, but a fast IO routine can _never_ write to an output variable and always have legal, well-defined behavior.